asgiref/sync.py

```diff
@@ -466,7 +468,7 @@ def thread_handler(self, loop, source_task, exc_info, func, *args, **kwargs):
         self.threadlocal.main_event_loop_pid = os.getpid()
         # Set the task mapping (used for the locals module)
         current_thread = threading.current_thread()
-        if AsyncToSync.launch_map.get(source_task) == current_thread:
+        if self.launch_map.get(current_thread):
```

Comment:
This is to stop us deleting `self.launch_map[current_thread]` twice for the same thread at the end of this function.

Generally `self.launch_map[current_thread]` here is supposed to look up the async task that spawned the current thread. But if the current thread is the one used by a `CurrentThreadExecutor` to run sync code in a wrapped async context, we can't guarantee its always called from the same async task (because it can be called from within asyncio.wait_for).

So... doing the check `== source_task` didn't work, but what we really care about is whether its safe to delete the launch map entry at the end of this function, i.e. are with the one that set it or not.